components/range_cache_memory_engine/src/background.rs

```diff
@@ -1091,7 +1019,29 @@ impl Runnable for BackgroundRunner {
                     let delete_range_scheduler = self.delete_range_scheduler.clone();
                     let core = self.core.clone();
                     let task = async move {
-                        core.evict_on_soft_limit_reached(&delete_range_scheduler);
+                        if let Some(range_stats_manager) = &core.range_stats_manager {
+                            let cached_region_ids =
+                                core.engine.read().range_manager().cached_regions();
+
+                            let evict_fn = |evict_region: &CacheRegion,
+                                            evict_reason: EvictReason,
+                                            cb: Option<Box<dyn FnOnce() + Send + Sync>>|
+                             -> Vec<CacheRegion> {
+                                let mut core = core.engine.write();
+                                core.mut_range_manager().evict_region(
```

Comment:
Will it block foreground write? If so, how long will it block?

Reply:
It will not block foreground operations.